### ISSUE TITLE
Infinite loop in undirected bidirectional()

### ISSUE THREAD

**Issue by dsbauer** [graphology/graphology-shortest-path#4]
```
var unweighted = require('../unweighted.js'),
    graphology = require('graphology'),
    UndirectedGraph = graphology.UndirectedGraph;

var net = new UndirectedGraph();

net.addNode(0);
net.addNode(1);
net.addEdge(0,1);

unweighted.bidirectional(net,0,1);  //-->infinite loop!
```
The problem manifests on line 142 of unweighted.js, but seems to be caused earlier by circularity in predecessor index: {0:'1', 1:'0'}.

**Comment by Yomguithereal** [comment]
Hello @dsbauer. I just released a v1.0.2 of the lib that should fix the issue. Can you check that it does and close the issue if this is indeed the case? Thanks for noticing the issue.

**Comment by dsbauer** [comment]
The original problem is fixed, thanks, but I'm seeing a new one, coming from the new code.
```
var net = new UndirectedGraph();

net.addNode(0);
net.addNode(1);
net.addNode(2);
net.addNode(3);
net.addEdge(0,1);
net.addEdge(0,2);
net.addEdge(1,2);
net.addEdge(2,3);

unweighted.bidirectional(net,0,3); // error
```
`TypeError: Cannot use 'in' operator to search for '#<Object>' in 2`
`at Function.bidirectional (unweighted.js:125:27)`

**Comment by Yomguithereal** [comment]
Just released v1.0.3. Hopefully this time everything works :).

**Comment by dsbauer** [comment]
That's better, thanks.  But here's another problem, which may be unrelated to recent changes: it returns some false positive paths when targeting unconnected nodes.  For example:
```
var net = new UndirectedGraph();

net.addNode(0);
net.addNode(1);
net.addNode(2);
net.addEdge(0,1);

unweighted.bidirectional(net,2,0); // --> null
unweighted.bidirectional(net,0,2); // --> ['0']
```
It seems like a greater variety of test networks could catch more errors.  I may be able to contribute later if you'd like help.  Thanks for the improvements!

**Comment by Yomguithereal** [comment]
Well that's embarassing :). Just released v1.0.4 with the fix. 

> It seems like a greater variety of test networks could catch more errors. I may be able to contribute later if you'd like help. Thanks for the improvements!

Sure, I'll be glad to have your help.
